File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/Common/Container.vb
Imports VecompSoftware.Helpers.ExtensionMethods

<Serializable()> _
Public Class Container
    Inherits AuditableDomainObject(Of Int32)
    Implements ISupportBooleanLogicDelete, ISupportRangeDelete

#Region " Fields "

    Private _protLocation As Location
    Private _protAttachLocation As Location

#End Region

#Region " Constructor "
    Public Sub New()
        ContainerGroups = New List(Of ContainerGroup)
        UniqueId = Guid.NewGuid()
    End Sub
#End Region

#Region " Properties "

    Overridable Property Name As String

    Overridable Property Note As String

    Overridable Property DocmLocation As Location

    Overridable Property ReslLocation As Location

    Overridable Property ProtLocation As Location
        Get
            Return _protLocation
        End Get
        Set(ByVal value As Location)
            _protLocation = value
        End Set
    End Property

    Overridable Property ProtAttachLocation As Location
        Get
            If _protAttachLocation Is Nothing Then
                Return _protLocation
            End If
            Return _protAttachLocation
        End Get
        Set(ByVal value As Location)
            _protAttachLocation = value
        End Set
    End Property

    Public Overridable Property IsActive As Boolean Implements ISupportBooleanLogicDelete.IsActive

    Overridable Property ContainerGroups As IList(Of ContainerGroup)

    Public Overridable Property ContainerProperties As IList(Of ContainerProperty)

    Public Overridable Property DocumentSeriesLocation As Location

    Public Overridable Property DocumentSeriesAnnexedLocation As Location

    ''' <summary> Abilita il rifiuto di protocollo. </summary>
    Public Overridable Property DocumentSeriesUnpublishedAnnexedLocation As Location

    ''' <summary></summary>
    ''' <remarks>Nato per ASL-TO2</remarks>
    Public Overridable Property Privacy As Short?

    ''' <summary></summary>
    ''' <remarks>Nato per ASL-TO2</remarks>
    Public Overridable Property HeadingFrontalino As String

    ''' <summary></summary>
    ''' <remarks>Nato per ASL-TO2</remarks>
    Public Overridable Property HeadingLetter As String

    Public Overridable Property Archive As ContainerArchive

    Public Overridable Property DeskLocation As Location

    Public Overridable Property UDSLocation As Location

    Public Overridable Property PrivacyLevel As Integer

    Public Overridable Property PrivacyEnabled As Boolean
#End Region

#Region " Methods "

    Public Overridable Function IsActiveRange() As Boolean Implements ISupportRangeDelete.IsActiveRange
        Return IsActive = True
    End Function

#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/Commons/Container.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Entity.DocumentArchives;
using VecompSoftware.DocSuiteWeb.Entity.DocumentUnits;
using VecompSoftware.DocSuiteWeb.Entity.Dossiers;
using VecompSoftware.DocSuiteWeb.Entity.Fascicles;
using VecompSoftware.DocSuiteWeb.Entity.OCharts;
using VecompSoftware.DocSuiteWeb.Entity.Protocols;
using VecompSoftware.DocSuiteWeb.Entity.Resolutions;
using VecompSoftware.DocSuiteWeb.Entity.Tenants;
using VecompSoftware.DocSuiteWeb.Entity.UDS;

namespace VecompSoftware.DocSuiteWeb.Entity.Commons
{
    public class Container : DSWBaseEntity
    {
        #region [ Constructor ]

        public Container() : this(Guid.NewGuid()) { }
        public Container(Guid uniqueId)
            : base(uniqueId)
        {
            Protocols = new HashSet<Protocol>();
            Resolutions = new HashSet<Resolution>();
            OChartItemContainers = new HashSet<OChartItemContainer>();
            ContainerGroups = new HashSet<ContainerGroup>();
            DocumentSeries = new HashSet<DocumentSeries>();
            DocumentUnits = new HashSet<DocumentUnit>();
            UDSRepositories = new HashSet<UDSRepository>();
            Dossiers = new HashSet<Dossier>();
            ContainerProperties = new HashSet<ContainerProperty>();
            Tenants = new HashSet<Tenant>();
            Fascicles = new HashSet<Fascicle>();
        }
        #endregion

        #region[ Properties ]

        public string Name { get; set; }
        public string Note { get; set; }
        public bool isActive { get; set; }
        public int? idArchive { get; set; }
        public byte? Privacy { get; set; }
        public string HeadingFrontalino { get; set; }
        public string HeadingLetter { get; set; }
        public int PrivacyLevel { get; set; }
        public bool PrivacyEnabled { get; set; }

        #region [fake properties]
        /// <summary>
        /// Proprietà fake per gestire l'inserimento automatico di gruppi
        /// </summary>
        public bool? AutomaticSecurityGroups { get; set; }
        /// <summary>
        /// Proprietà fake per gestire l'inserimento automatico di gruppi
        /// </summary>
        public string PrefixSecurityGroupName { get; set; }
        /// <summary>
        /// Proprietà fake per gestire l'inserimento automatico di gruppi
        /// </summary>
        public ContainerType? ContainerType { get; set; }
        /// <summary>
        /// Proprietà fake per gestire l'inserimento automatico in SecurityUser
        /// </summary>
        public string SecurityUserAccount { get; set; }
        /// <summary>
        /// Proprietà fake per gestire l'inserimento automatico in SecurityUser
        /// </summary>
        public string SecurityUserDisplayName { get; set; }
        #endregion

        #endregion

        #region [ Navigation Properties ]
        public virtual Location DocmLocation { get; set; }
        public virtual Location ProtLocation { get; set; }
        public virtual Location ReslLocation { get; set; }
        public virtual Location DeskLocation { get; set; }
        public virtual Location UDSLocation { get; set; }
        public virtual Location ProtAttachLocation { get; set; }
        public virtual Location DocumentSeriesLocation { get; set; }
        public virtual Location DocumentSeriesUnpublishedAnnexedLocation { get; set; }
        public virtual Location DocumentSeriesAnnexedLocation { get; set; }
        public virtual ICollection<Protocol> Protocols { get; set; }
        public virtual ICollection<Resolution> Resolutions { get; set; }
        public virtual ICollection<OChartItemContainer> OChartItemContainers { get; set; }
        public virtual ICollection<ContainerGroup> ContainerGroups { get; set; }
        public virtual ICollection<DocumentSeries> DocumentSeries { get; set; }
        public virtual ICollection<DocumentUnit> DocumentUnits { get; set; }
        public virtual ICollection<UDSRepository> UDSRepositories { get; set; }
        public virtual ICollection<Dossier> Dossiers { get; set; }
        public virtual ICollection<ContainerProperty> ContainerProperties { get; set; }
        public virtual ICollection<Tenant> Tenants { get; set; }
        public virtual ICollection<Fascicle> Fascicles { get; set; }
        public virtual ICollection<CategoryFascicleRight> CategoryFascicleRights { get; set; }
        #endregion
    }
}
